inference-engine/src/mkldnn_plugin/mkldnn_graph_optimizer.cpp

```diff
@@ -1530,15 +1530,20 @@ void MKLDNNGraphOptimizer::FuseEltwiseAndSimple(MKLDNNGraph &graph) {
                     }
                 } else {
                     MKLDNNEdgePtr &remEdge = p_edge;
+                    auto parentEltwise = parentNode;
                     int inNum = 0;
+                    int outNum = parentEltwise->getParentEdges().size();
                     if (remEdge) {
                         inNum = remEdge->getInputNum();
+                        // Need to keep order for MulAdd
+                        if (childNode->getAlgorithm() == EltwiseMulAdd) {
```

Comment:
Do we need this condition. Won't it work for all eltwise types?

Reply:
That does not work for other algorithms.